pyaqueduct/plugin.py

```diff
@@ -0,0 +1,74 @@
+""" The module contains classes representing interface with plugins. """
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+from pyaqueduct.client import AqueductClient
+from pyaqueduct.client.plugin_types import (PluginData,
+                                            PluginExecutionResultData,
+                                            PluginFunctionData,
+                                            PluginParameterData)
+
+
+class PluginExecutionResult(PluginExecutionResultData):
+    """Class representing result of plugin execution"""
+
+
+class PluginParameter(PluginParameterData):
+    """Plugin parameter class"""
+
+
+class PluginFunction(PluginFunctionData):
```

Comment:
This inheritance is going to be problematic. You are inheriting from a data class whose purpose is only data transfer. Instead, assign a member variable of this type. You can look at the `Experiment` module.